### PR TITLE
[Styles] Render parking spaces

### PR DESCRIPTION
**Pull request by RedAuburn**
(needs map regeneration)
finally figured out how to generate maps 🥳

renders parking spaces. I tried to keep it unobtrusive which is why it's hard to see, but i think this is ok as rendering them strongly would make the map look cluttered and overly complex.

link to MWM with parking_spaces generated: https://drive.google.com/file/d/1kkbFmjlqddTLOvJx2xGCF_wi6xaV5KOf
location: 51.4986905, 0.0039843

z18
Screenshot 2022-05-14 at 15 17 58
>

z19 onwards
Screenshot 2022-05-14 at 15 18 13

Signed-off-by: Harry Bond <endim8@pm.me>

### CODE REVIEW

**Comment by j5k** [comment]
Looks good, I like it

**Review by biodranik — COMMENTED** [review]
Does it really make sense on Z18? Especially on smaller screens?

**Comment by endim8** [comment]
> Does it really make sense on Z18? Especially on smaller screens?

I'm not sure, I'll try to build with the mwm and test it on my phone.

Z18 sets the opacity to half of what it is at z19+, and is more of a fade out to stop it popping in, but other small things appear at z18 (bench icons, bins etc) so I thought it'd be the right level.

I'll do further testing 👍

File: data/styles/clear/include/Basemap.mapcss [hunk 1]
```diff
@@ -577,10 +578,16 @@ area|z15-[landuse=churchyard]
 area|z10-[landuse=farmland],
 {fill-position: background; fill-color: @farmland; fill-opacity: 1;}
 
+
 /* Parking */
 
 area|z15-[amenity=parking]
 {fill-color: @parking;}
+line|z18-[amenity=parking_space]
+{fill-position: background;width: 2;color: @white;opacity: 0.3;}
```
Comment (vng) [inline data/styles/clear/include/Basemap.mapcss:587]: fill-position: background; is useless here, as parking_space become linear.

**Review by vng — CHANGES_REQUESTED** [review]
Remove all amenity=parking_space entries from vehicle/include/Basemap.mapcss
They are useless in a new concept, but resulting drules_proto.txt still contains z15 styles.

File: data/styles/clear/include/Basemap.mapcss (hunk 1 above)
Comment (endim8) [reply data/styles/clear/include/Basemap.mapcss]: fixed

**Comment by endim8** [comment]
> Remove all amenity=parking_space entries from vehicle/include/Basemap.mapcss They are useless in a new concept, but resulting drules_proto.txt still contains z15 styles.

I'd completely forgotten about this PR, fixed 👍

**Comment by endim8** [comment]
`FAILED
2022-06-04 02:32:34,102 - osm_auth_tests`
 
I know osm.org is in the process of implementing OAuth 2.0, so this failure is likely unrelated?

**Comment by biodranik** [comment]
@vng will it slow down the rendering in cities?

**Comment by vng** [comment]
Well, I thought a bit about this PR and still can't figure out, do I like it or not :)
Nice picture of a bit parking area vs useless rendering info on a map ..

According to the taginfo, there are a lot of amenity=parking_space AND capacity=1 (~600K, out of 1300K of all parking spaces). And I suspect that such kind of isolated 1-place parking spaces are not tagged as general amenity=parking.

Probably, better to figure out such isolates spaces and draw them as small parking icon?

**Comment by pastk** [comment]
What are the practical use cases? (except for an eye candy)

I guess its useful for the motorists
- to understand the positioning of the spaces, e.g. angled / parallel parking, etc. and prepare accordingly before getting to the place
- to understand if a long vehicle (e.g. with a trailer) will fit
- to see if there are special places available, e.g. for disabled (need to support [`parking_space=*`](https://wiki.openstreetmap.org/wiki/Key:parking_space))

**Comment by endim8** [comment]
How about this? If performance is a real concern, there could be a generator stage to merge all connected `amenity=parking_space` areas into one. I think this style nicely complements the new disabled parking space icons.
fgfegfv


@pastk, honestly, mainly for eye candy. It could be useful for orientation on foot though, and shows the detail of OSM data.

**Comment by biodranik** [comment]
Some levels of details are _too detailed_ :)

**Comment by endim8** [comment]
> Some levels of details are _too detailed_ :)

Aww, alright. Fair enough 🥲

**Comment by mpawelski** [comment]
☹

I actually draw couple of `amenity=parking_space` around my place in anticipation to see them in OMaps when thin PR gets merged (the fact that this tag is rendered in osm-carto was not motivating enough for me to add it earlier😅)

Is this PR really "rejected" or @endim8 closed it prematurely? Because all the examples provided by @endim8 really looks like an improvement to me. 😍

> What are the practical use cases? (except for an eye candy)

>@pastk, honestly, mainly for eye candy. It could be useful for orientation on foot though, and shows the detail of OSM data.

I agree. This is mostly to make map prettier and more detailed in high zoom. But is this justification not good enough? IMO small style changes like this can really make a difference in making OMaps a better (prettier and more detailed) app.

If it's not affecting performance then I see no reason to reject this change.
But of course the decision is up to the core team if they agree with me. This whole post is just my opinion 😉

**Comment by IronMaltese** [comment]
I thought this PR looks really nice too, if this was rejected due to storage there should be an option to exclude smaller details like park benches and fountains from downloading

If this is due to performance, low power mode should hide parking spaces, otherwise I would love to see this implemented!

**Comment by endim8** [comment]
I'm glad people would like it, I would too 😄

**Comment by vng** [comment]
I don't mind here. Especially if this _waffle grid_ will make you folks happy :)

Should we make special processing (a small icon?) of amenity=parking_space AND capacity=1 (~600K, out of 1300K of all parking spaces), as I mentioned above?

**Comment by mpawelski** [comment]
BTW, I checked how OM and maps.me render such `amenity=parking_space` without `amenity=parking`. I checked it for one example because other were too new for maps.me (this is what happens when you update maps once a year 😆). Conslusions:
- OM is not rendering it. 
- maps.me render parking icon without any background

And I don't want maps.me to be more detailed than OM so maybe we should consider such edge case and render such parking spaces like parking. Even if it's a mapping error.

[example 2](https://www.openstreetmap.org/way/937810117)
Screenshot_20220829-233729
Screenshot_20220829-233922

**Comment by biodranik** [comment]
Let's bring some _real_ value to this change! In Switzerland, for example, there are blue parking zones that allow a free 1-hour stay. There are also white zones (paid, with limited max time). It would be great if you can collect information about colors/parking types in different countries and incorporate them into this change. Then it would be easier for drivers to see the parking type on the map directly based on its color!

**Comment by endim8** [comment]
> Let's bring some _real_ value to this change! In Switzerland, for example, there are blue parking zones that allow a free 1-hour stay. There are also white zones (paid, with limited max time).

I'm struggling to find any examples for countries other than Switzerland, it might also be confusing as parking spaces are often painted blue
it might be better to just do standard parking spaces for now :)

**Comment by mpawelski** [comment]
>I'm struggling to find any examples for countries other than Switzerland, it might also be confusing as parking spaces are often painted blue

Yeah, collecting info how different parking space types from different countries might be tough. 

>In Switzerland, for example, there are blue parking zones that allow a free 1-hour stay. There are also white zones (paid, with limited max time).

If I remember correctly, in Italy it was the opposite :D Blue lines paid (except at night), white lines usually free (but sometimes limited to max 1h). 

But the more important question is how it's mapped in OSM? On `amenity=parking` or `amenity=parking_space`? I checked the wiki and I'm not sure if we have that detailed info about parking spaces in OSM.

Info in [parking_space](https://wiki.openstreetmap.org/wiki/Key:parking_space) tag looks pretty limited. Usually people just map its capacity and info if it's for disabled people,  ([taginfo](https://taginfo.openstreetmap.org/tags/amenity=parking_space#combinations)). Sometimes they also map if it's free and for how long, but `access`, `fee` and `maxstay` tags are more popular on parent `amenity=parking` tag ([taginfo](https://taginfo.openstreetmap.org/tags/amenity=parking#combinations)) anyway.

>it might be better to just do standard parking spaces for now :)

I agree, and TBH probably we won't go much further than that in future too, because of limited OSM data we have about parking spaces ;)

**Comment by biodranik** [comment]
Let's create an issue to track and render later colored parking areas. Maybe someone can also propose OpenStreetMap.org tagging for it.

**Pull request closed without merging**